docs/overview/Lifecycle.md

```diff
@@ -0,0 +1,335 @@
+If a module failed to load for any reason or when validating, One App will
+log the error and recover gracefully, adding the module to a `block` list until
+updated. Holocron modules are reloaded whenever the module map is polled and does
+not match what is on the server.
+
+After the modules are loaded and validated, One App can start accepting requests.
+Once One App becomes operational, it begins to periodically poll the module map
+URL supplied by the user to observe for any changes. If a Holocron module is added,
+updated or removed from the module map, changed modules will be re-installed into
+memory.
+
+## Runtime App Configuration
+
+As we discussed in the [Getting Started Guide](../Getting-Started.md), each
+Holocron module can supply an [`appConfig`](../api/modules/App-Configuration.md)
+that is executed on and configures the server.
+
+An example `Module.appConfig`:
+
+```jsx
+import React from 'react';
+
+export default function RootModule({ children }) {
+  return children;
+}
+
+if (!global.BROWSER) {
+  RootModule.appConfig = {
+    csp: 'default-src \'self\'',
+  };
+}
+```
+
+We can use this additional configuration interface to further tweak
+One App to our needs. The only requirement is that the root module
+needs to be configured with a valid CSP.
+
+## Server Side Rendering
+
+When One App has loaded all of our modules and started the `express` server,
+the app becomes ready to render our Holocron modules. There are a few ways that
+One App can render our Holocron modules:
+
+- default render path, using the request URL to render the modules that match
+- partial renders, renders only the selected Holocron module in isolation
+
+We will cover both rendering modes and step through what happens when One App
+gets a request. We can use `Module.appConfig` to configure key aspects for server
+side rendering (SSR) in One App to fit our use cases.
+
+### Rendering Life Cycle
+
+Once a request reaches the One App server, the app begins the render cycle
+by assigning various headers to the response before the html is rendered.
+
+**Routing**
+
+One App [creates a Holocron Redux store](create-holocron-store) per request
+then matches the request URL to determine which modules to render. The route is matched using
+[`@americanexpress/one-app-router`][one-app-router] and composes all the [`ModuleRoute`s](holocron-module-route) and
+[`Route`s](one-app-router-route) as render props for the [`Router`](one-app-router-router).
+
+```jsx
+import React from 'react';
+import ModuleRoute from 'holocron-module-route';
+
+export default function MyModule({ children }) {
+  return children;
+}
+
+MyModule.holocron = {
+  name: 'my-module',
+};
+
+MyModule.childRoutes = () => [
+  <ModuleRoute moduleName="my-module">
+    <ModuleRoute moduleName="my-layout-module">
+      <ModuleRoute moduleName="landing-page-module" path="/" />
+    </ModuleRoute>
+  </ModuleRoute>,
+];
+```
+
+**Data Loading**
```

Comment:
What do you think a small mention of fetchye and a link to the recipe?

Reply:
The lifecycle docs was intended to focus on One App life cycles (server, holocron, render) and the configuration that drives it - Would mentioning `fetchye` fit better in the `Getting Started` guide? Or do you think that would benefit the topic and keep it focused?

Reply:
agreed with @Francois-Esquire lets keep this focused on the lifecycle / flow